src/organizations/components/DeleteOrgOverlay.tsx

```diff
@@ -18,22 +18,80 @@ import {
   JustifyContent,
   Overlay,
 } from '@influxdata/clockface'
+import {track} from 'rudder-sdk-js'
 
 // Utils
 import {deleteAccount} from 'src/client/unityRoutes'
 import {notify} from 'src/shared/actions/notifications'
 import {accountSelfDeletionFailed} from 'src/shared/copy/notifications'
+import CancellationReasonsDropdown from './CancellationReasonsForm'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {event} from 'src/cloud/utils/reporting'
+
+// Selectors
+import {getQuartzMe} from 'src/me/selectors'
+import {DeleteOrgContext} from './DeleteOrgContext'
+import {getOrg} from 'src/organizations/selectors'
 
 const DeleteOrgOverlay: FC = () => {
   const history = useHistory()
   const [hasAgreedToTerms, setHasAgreedToTerms] = useState(false)
   const dispatch = useDispatch()
+  const [formErrors, setFormErrors] = useState({})
+  const {shortSuggestion, isShortSuggestionEnabled, suggestions} = useContext(
+    DeleteOrgContext
+  )
+  const quartzMe = useSelector(getQuartzMe)
+  const org = useSelector(getOrg)
 
   const handleClose = () => {
     history.goBack()
   }
 
+  const isFormValid = () => {
+    if (!isFlagEnabled('trackCancellations')) {
+      return true
+    }
+
+    const errors = {}
+    let hasErrors = false
+    if (isShortSuggestionEnabled && !shortSuggestion) {
+      errors['shortSuggestion'] = 'Please enter a suggestion'
+      hasErrors = true
+    }
+
+    if (!suggestions) {
+      errors['suggestions'] = 'Please enter suggestions'
+      hasErrors = true
+    }
+
+    setFormErrors(errors)
+
+    return !hasErrors
+  }
```

Comment:
wondering if we should memoize this

Reply:
Memoizing this. This does change the behavior of the form. Instead of evaluating on click, it's now evaluating on every change. Thoughts?
![Kapture 2021-10-04 at 18 30 52](https://user-images.githubusercontent.com/1637395/135933420-d8a2d295-ae9d-422c-a053-10052f7ec1c1.gif)

Reply:
i like this more. :)